### PR TITLE
Hw6

### PR DESCRIPTION
**Pull request by iRezograf**
Please aprove it.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -139,6 +147,12 @@
             <version>RELEASE</version>
             <scope>compile</scope>
         </dependency>
+        <dependency>
+            <groupId>org.testng</groupId>
+            <artifactId>testng</artifactId>
+            <version>7.4.0</version>
+            <scope>compile</scope>
+        </dependency>
```
Comment (shoggy) [inline pom.xml:154]: duplicated dependency

File: src/test/java/ru/training/at/hw6/interfaces/TestsInit.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
+package ru.training.at.hw6.interfaces;
+
+
+import com.epam.jdi.light.driver.WebDriverUtils;
+import com.epam.jdi.light.elements.init.PageFactory;
+import org.testng.annotations.AfterSuite;
+import org.testng.annotations.BeforeSuite;
+import ru.training.at.hw6.SiteJdi;
+
+public interface TestsInit {
```
Comment (shoggy) [inline src/test/java/ru/training/at/hw6/interfaces/TestsInit.java]: Initially interface is not intended for such useage

File: src/main/java/ru/training/at/hw6/providers/JsonDataParser.java [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+        Object[][] data = new Object[countOfRootGets][dataNumberFieldInForm];
+
+        try (FileReader reader = new FileReader(
+                "src/test/resources/JDI_ex8_metalsColorsDataSet.json")) {
+
+            JSONObject rootJsonObj = (JSONObject) parser.parse(reader);
+
+            for (int i = 0; i < countOfRootGets; i++) {
+
```
Comment (shoggy) [inline src/main/java/ru/training/at/hw6/providers/JsonDataParser.java]: Input data parsing depends not only on the data itself, but also on external config values

**Review by shoggy — CHANGES_REQUESTED** [review]

File: src/test/java/ru/training/at/hw6/JdiTests.java [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    }
+
+    public void loginTest(LoginUserData loginUserData) {
+        SiteJdi.open();
+        homePage.userIcon.click();
+        homePage.name.sendKeys(loginUserData.getUserName());
+        homePage.password.sendKeys(loginUserData.getPassword());
+        homePage.loginButton.click();
+        homePage.userName.assertThat().is().text(loginUserData.getUserNameAfterLogged());
+    }
```
Comment (shoggy) [inline src/test/java/ru/training/at/hw6/JdiTests.java]: This should be implemented using `com.epam.jdi.light.elements.composite.Form`

File: pom.xml (hunk 1 above)
Comment (iRezograf) [reply pom.xml:154]: done

File: src/test/java/ru/training/at/hw6/interfaces/TestsInit.java (hunk 2 above)
Comment (iRezograf) [reply src/test/java/ru/training/at/hw6/interfaces/TestsInit.java:10]: Replaced by:
public class JdiTests extends TestsInit{

Comment (iRezograf) [reply src/test/java/ru/training/at/hw6/interfaces/TestsInit.java]: Included into^
public class JdiTests {
    LoginUserData loginUserData;

    @BeforeSuite(alwaysRun = true)
    static void beforeSuite() {
        PageFactory.initElements(SiteJdi.class);
    }

    @AfterSuite(alwaysRun = true)
    static void afterSuite() {
        WebDriverUtils.killAllSeleniumDrivers();
    }

File: src/test/java/ru/training/at/hw6/JdiTests.java (hunk 4 above)
Comment (iRezograf) [reply src/test/java/ru/training/at/hw6/JdiTests.java]: done

File: src/main/java/ru/training/at/hw6/providers/JsonDataParser.java (hunk 3 above)
Comment (iRezograf) [reply src/main/java/ru/training/at/hw6/providers/JsonDataParser.java]: done

File: src/main/java/ru/training/at/hw6/forms/LoginForm.java [hunk 5]
```diff
@@ -0,0 +1,39 @@
+package ru.training.at.hw6.forms;
+
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.light.ui.html.elements.common.Button;
+import com.epam.jdi.light.ui.html.elements.common.Link;
+import com.epam.jdi.light.ui.html.elements.common.Text;
+import com.epam.jdi.light.ui.html.elements.common.TextField;
+import ru.training.at.hw6.entities.User;
+
+public class LoginForm {
```
Comment (shoggy) [inline src/main/java/ru/training/at/hw6/forms/LoginForm.java]: Form should extend Form

File: src/main/java/ru/training/at/hw6/forms/LoginForm.java [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+        if (user.getName() != null) {
+            name.clear();
+            name.sendKeys(user.getName());
+        }
+        if (user.getPassword() != null) {
+            password.clear();
+            password.sendKeys(user.getPassword());
+        }
+        loginButton.click();
+    }
```
Comment (shoggy) [inline src/main/java/ru/training/at/hw6/forms/LoginForm.java:38]: please refer to `com.epam.jdi.light.elements.composite.Form#login(T)`

File: src/main/java/ru/training/at/hw6/forms/MetalsAndColorsForm.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+import com.epam.jdi.light.elements.complex.Checklist;
+import com.epam.jdi.light.elements.complex.WebList;
+import com.epam.jdi.light.elements.complex.dropdown.Dropdown;
+import com.epam.jdi.light.elements.pageobjects.annotations.FindBy;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.JDropdown;
+import com.epam.jdi.light.elements.pageobjects.annotations.locators.UI;
+import com.epam.jdi.light.ui.html.elements.common.Button;
+import com.epam.jdi.light.ui.html.elements.complex.RadioButtons;
+
+public class MetalsAndColorsForm  {
```
Comment (shoggy) [inline src/main/java/ru/training/at/hw6/forms/MetalsAndColorsForm.java:13]: This class should extend Form

File: src/test/java/ru/training/at/hw6/JdiTests.java [hunk 8]
```diff
@@ -0,0 +1,114 @@
...
+        for (String val : metalsAndColors.getVegetables()) {
+            MetalsAndColorsForm.vegetables.select(val);
+        }
+
+        for (String val : metalsAndColors.getElements()) {
+            MetalsAndColorsForm.elements.select(val);
+        }
+
+        MetalsAndColorsForm.submitButton.click();
+    }
```
Comment (shoggy) [inline src/test/java/ru/training/at/hw6/JdiTests.java]: This method should be in corresponding form class, 
please look for `com.epam.jdi.light.elements.composite.Form#fill(T)` or `com.epam.jdi.light.elements.composite.Form#submit(T)`

**Review by shoggy — CHANGES_REQUESTED** [review]

File: src/main/java/ru/training/at/hw6/forms/LoginForm.java (hunk 5 above)
Comment (iRezograf) [reply src/main/java/ru/training/at/hw6/forms/LoginForm.java]: done

File: src/main/java/ru/training/at/hw6/forms/LoginForm.java (hunk 6 above)
Comment (iRezograf) [reply src/main/java/ru/training/at/hw6/forms/LoginForm.java:38]: refered but didn't find something wrong

File: src/test/java/ru/training/at/hw6/JdiTests.java (hunk 8 above)
Comment (iRezograf) [reply src/test/java/ru/training/at/hw6/JdiTests.java]: done

**Review by shoggy — APPROVED** [review]

**Pull request merged**